_episodes/l6-06-Best_practices.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "General unit testing best practice"
+teaching: TBC
+exercises: TBC
+questions:
+- "What are some best practices that should be followed when writing unit tests?"
+objectives:
+- "Understand some of the best practices for writing unit tests."
+keypoints:
+- "Unit tests should be isolated"
+- "Unit tests should be repeatable"
+- "Unit tests should be readable"
+- "Unit tests should have one assertion per test"
+---
+
+## 1. Best Practices for writing unit tests
+
+Below, we share some of the best practices that should be followed while writing unit tests. The suggestions below are general in nature and apply to all testing frameworks, not just C++. Some of the suggestions are from the book [Modern C++ programming with Test-Driven-Development](`https://www.amazon.co.uk/Modern-Programming-Test-Driven-Development-Better/dp/1937785483`) by *Jeff Langr*. We have also used some suggestions that came out from a simple Google search and those have not been explicitly cited.
```

Comment:
I probably wouldn't admit to the Google search part! ^^

Reply:
I agree.